pkg/controllers/dashboardapi/settings/settings_test.go

```diff
@@ -261,3 +278,181 @@ func populateTestCases() []*testCase {
 
 	return testCases
 }
+
+func TestSetAllWithDefaultOnUpgrade(t *testing.T) {
```

Comment:
Nit: We've discussed this before, but I'm not a fan of creating separate test functions for the same function under test. It makes coverage harder to assess, and makes it harder to maintain over time.

I'm marking this as a Nit since I think long term we may want something different for these tests, so I don't think this needs to be 100% perfect for this implementation.